File: 양승민/입문/src/lec04/lec04Main.java
package lec04;

public class lec04Main {
    public static void main(String[] args) {

        // 1. 비교 연산자
        Integer a = 5;
        Integer b = 5;

        if (a == b) {
            System.out.println("동일성 비교");
        } else if (a.equals(b)) {
            System.out.println("동등성 비교");
        }

        // 3. 연산자 오버로딩
        JavaMoney money1 = new JavaMoney(1_000L);
        JavaMoney money2 = new JavaMoney(2_000L);
        System.out.println(money1.plus(money2));

    }
}

File: 양승민/입문/src/lec04/lec04Main.kt
package lec04

fun main() {

    // 1. 비교 연산자
    val a = 5;
    val b = 5;

    if (a === b) {
        println("동일성 비교")
    } else if (a == b) {
        println("동등성 비교")
    }

    // 2. in 연산자
    val str = "ABCDE"
    if ('B' in str) {
        println("str 안에 B 있음")
    }
    if ('H' !in str) {
        println("str 안에 H 없음")
    }

    // 3. 연산자 오버로딩
    val money1 = Money(1_000L)
    val money2 = Money(2_000L)
    println(money1 + money2)
}